framework_tvb/tvb/interfaces/web/controllers/simulator_controller.py

```diff
@@ -780,7 +780,8 @@ def set_simulation_length(self, **data):
         is_simulator_load = common.get_from_session(common.KEY_IS_SIMULATOR_LOAD) or False
         session_burst_config = common.get_from_session(common.KEY_BURST_CONFIG)
 
-        next_form = SimulatorFinalFragment()
+        simulation_number = self.burst_service.get_available_bursts(common.get_current_project().id).__len__() + 1
```

Comment:
I suggest using here a count query. This would become a performance problem (to load all bursts and just count).
Have you tested what is happening when there are no simulations?